devicetypes/smartthings/cooper-scene-controller.src/cooper-scene-controller.groovy

```diff
@@ -0,0 +1,268 @@
+	//As this process will take a while, we use controller's LED indicators to display progress.
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: 0).format())
+	cmds << new physicalgraph.device.HubAction(zwave.associationV1.associationGet(groupingIdentifier:1).format())
+	cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfGet(groupId:1).format())
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: 1).format())
+	cmds << new physicalgraph.device.HubAction(zwave.associationV1.associationGet(groupingIdentifier:2).format())
+	cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfGet(groupId:2).format())
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: 3).format())
+	cmds << new physicalgraph.device.HubAction(zwave.associationV1.associationGet(groupingIdentifier:3).format())
+	cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfGet(groupId:3).format())
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: 7).format())
+	cmds << new physicalgraph.device.HubAction(zwave.associationV1.associationGet(groupingIdentifier:4).format())
+	cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfGet(groupId:4).format())
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: 15).format())
+	cmds << new physicalgraph.device.HubAction(zwave.associationV1.associationGet(groupingIdentifier:5).format())
+	cmds << new physicalgraph.device.HubAction(zwave.sceneControllerConfV1.sceneControllerConfGet(groupId:5).format())
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: 31).format())
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorSet(value: 0).format())
+	//Make sure cloud is in sync with device
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorGet().format())
+	if (!getDataValue("manufacturer")) {
+		cmds << new physicalgraph.device.HubAction(zwave.manufacturerSpecificV2.manufacturerSpecificGet().format())
+	}
+	//Long interval to make it possible to process association set commands if necessary
+	sendHubCommand(cmds, 3000)
+}
+
+def refresh() {
+	def cmds = []
+	// when using All Off feature, controller will often not respond to Indicator Get command
+	// using delayed refresh proves more reliable in this case
+	cmds << new physicalgraph.device.HubAction("delay 500")
+	cmds << new physicalgraph.device.HubAction(zwave.indicatorV1.indicatorGet().format())
+	//Indicator returns number which is a bit representation of current state of switches
+	sendHubCommand(cmds, 500)
+}
+
+def parse(String description) {
+	if (description.startsWith("Err")) {
+		if (description.startsWith("Err 106") && !state.sec) {
+			state.sec = 0
```

Comment:
Can this device ever add securely? My guess is not based on none of the other commands being encapsulated. Since that's the case this check seems unnecessary.

Reply:
No, it can not.

I missed that. This was left from template. Thanks for the catch.

Will remove this.